test/test_core.py

```diff
@@ -4177,6 +4179,21 @@ def test_dylink_safe_heap(self):
     self.set_setting('SAFE_HEAP')
     self.do_basic_dylink_test()
 
+  @needs_dylink
+  def test_dylink_locate_file(self):
+    name = 'liblib.so'
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
I don't think there is any need to use a random tmpdir here.

How about just doing:`

```
so_dir = 'so_dir'
os.makedir(so_dir)
```

Each test already runs inside its own temporary sandbox directory anyway

Reply:
Sounds good to me. I was trying to clean up after myself but I can see now that tests run in a tmpdir anyway.